backend/write_to_catalog/task.py

```diff
@@ -0,0 +1,57 @@
+import boto3
+from pystac import STAC_IO, Catalog, CatalogType  # type: ignore[import]
+
+from ..api_responses import BODY_KEY
+from ..pystac_io_methods import read_method, write_method
+from ..resources import ResourceName
+from ..types import JsonObject
+
+STAC_IO.write_text_method = write_method
+STAC_IO.read_text_method = read_method
+
+S3_CLIENT = boto3.client("s3")
+
+ROOT_CATALOG_ID = "root_catalog_id"
+ROOT_CATALOG_TITLE = "Geospatial Datalake Root Catalog"
+ROOT_CATALOG_DESCRIPTION = "The root catalog which links to all datasets catalogs in the data lake"
+CATALOG_KEY = "catalog.json"
+
+
+def lambda_handler(_event: JsonObject, _context: bytes) -> JsonObject:
+    """Main Lambda entry point."""
+
+    results = S3_CLIENT.list_objects(
+        Bucket=ResourceName.STORAGE_BUCKET_NAME.value, Prefix=CATALOG_KEY
+    )
+
+    # create root catalog if it doesn't exist
+    if "Contents" not in results:
```

Comment:
Since you have an `else` clause it would probably be easier to read this if you flip the clauses around to `if "Contents" in results:`. It basically avoids a double negative in the `else`.

Also, the "Contents" literal is used in two places in the code, here and in a test, so we should probably pull out a constant.